bookmarks/views.py

```diff
@@ -36,41 +36,49 @@
 import json
 
 def bookmarks(request, username, category_id = None):
-    
+
     user = get_object_or_404(User, username__iexact=username)
     is_owner = request.user.is_authenticated and user == request.user
-    
+
     '''
     if is_owner and request.POST and "create_cat" in request.POST:
         form_bookmark_category = BookmarkCategoryForm(request.POST, instance=BookmarkCategory(user=user))
         if form_bookmark_category.is_valid():
             form_bookmark_category.save()
-        
+
         form_bookmark_category = BookmarkCategoryForm()
-        
+
     form_bookmark_category = BookmarkCategoryForm()
-    '''    
-    
+    '''
+
     n_uncat = Bookmark.objects.select_related("sound").filter(user=user,category=None).count()
-    
+    category = None
+
     if not category_id:
         bookmarked_sounds = Bookmark.objects.select_related("sound").filter(user=user,category=None)
     else:
         category = get_object_or_404(BookmarkCategory,id=category_id,user=user)
         bookmarked_sounds = category.bookmarks.select_related("sound").all()
-    
+
     bookmark_categories = BookmarkCategory.objects.filter(user=user)
-    
-    return render_to_response('bookmarks/bookmarks.html', combine_dicts(locals(),paginate(request, bookmarked_sounds, 30)), context_instance=RequestContext(request))
+
+    tvars = {
+        'bookmark_categories': bookmark_categories,
+        'category': category,
+        'n_uncat': n_uncat,
+        'user': user,
+    }
+    tvars = combine_dicts(tvars, paginate(request, bookmarked_sounds, 30))
```

Comment:
combine_dicts should really only be used in functional chains. As { }.update() doesn't return anything it can't be used in that way, but in this case you don't need it...